### PR TITLE
Add some rust doc comments to exported things from rust/src/config

### PR DESCRIPTION
**Pull request by selaux**
I really liked the adding of doc comments in #820, so I added comments to all exported things in `rust/src/config`. This should give a better overview over what exists in rust code and the responsibilities of the individual modules.

To see the current docs you can run `cargo doc && firefox target/doc/stracciatella/index.html` in the `rust` subdirectory (adapt for your favorite other browser :wink:)

### CODE REVIEW

File: rust/src/config/ja2_json.rs [hunk 1]
```diff
@@ -36,17 +37,19 @@ fn build_json_config_location(stracciatella_home: &PathBuf) -> PathBuf {
...
 
 impl Ja2Json {
+    /// Construct a Ja2Json instance from the stracciatella home directory
     pub fn from_stracciatella_home(stracciatella_home: &PathBuf) -> Self {
         let path = build_json_config_location(stracciatella_home);
         Ja2Json { path }
     }
 
-    pub fn get_content(&self) -> Result<Ja2JsonContent, String> {
+    fn get_content(&self) -> Result<Ja2JsonContent, String> {
```
Comment (lynxlynxlynx) [inline rust/src/config/ja2_json.rs]: was this change intentional?

File: rust/src/config/engine_options.rs [hunk 2]
```diff
@@ -4,21 +4,36 @@ use crate::Resolution;
 use crate::ScalingQuality;
 use crate::VanillaVersion;
 
+/// Struct that is used to store the engines configuration parameters
 #[derive(Debug, PartialEq)]
 pub struct EngineOptions {
+    /// Path to configuration directory in the users home directory
```
Comment (lynxlynxlynx) [inline rust/src/config/engine_options.rs]:
```suggestion
    /// Path to configuration directory in the user's home directory
```

File: rust/src/config/stracciatella_home.rs [hunk 3]
```diff
@@ -1,6 +1,7 @@
 use dirs;
 use std::path::PathBuf;
 
+/// Find ja2 stracciatella configuration directory inside the users home directory
```
Comment (lynxlynxlynx) [inline rust/src/config/stracciatella_home.rs]:
```suggestion
/// Find ja2 stracciatella configuration directory inside the user's home directory
```

File: rust/src/config/ja2_json.rs (hunk 1 above)
Comment (selaux) [reply rust/src/config/ja2_json.rs]: Yes, the function is not publicly used, so I decided to make it not public. Should have added that to a separate commit.
